core/tests/puppeteer-acceptance-tests/blog-post-draft-blogs.js

```diff
@@ -0,0 +1,32 @@
+const acceptanceTests = require("./utility-functions/puppeteer_utils.js");
+const testConstants = require("./utility-functions/testConstants.js");
+
+
+const blogEditOptions = "e2e-test-blog-post-edit-box";
+
+async function deleteDraftAsBlogAdmin() {
+  const user = await new acceptanceTests();
+  const page = await user.init();
+
+  await user.goto(testConstants.URLs.home);
+  await user.clickOn("button", "OK");
+  await user.clickOn("span", "Sign in");
+  await user.type(testConstants.SignInDetails.inputField, "testadmin@example.com");
+  await user.clickOn("span", "Sign In");
+  
+  await user.goto(testConstants.URLs.BlogDashboard, testConstants.Dashboard.MainDashboard);
+
+  try{
+    await user.clickOn("button", blogEditOptions); // an icon
```

Comment:
All these should become blogPostAdmin.deleteBlogPost(...) where the '...' is something that identifies the blog post visually on the page.

Basically, you'll want each "sentence" in this file to be something that a product manager might use to describe what's going on. They won't be talking at the level "click this button", etc. -- it'll be more at the conceptual level, like "log in", "create new post with this content", "delete this post", etc.

Reply:
by this concept, for each test should we create a class containing such functions like logIn(), uploadAudio(), createNewBlog(), pubishBlog(), etc.. but again where we will be this class and from where we will be calling those functions?